src/org/usfirst/frc/team467/robot/Elevator.java

```diff
@@ -0,0 +1,163 @@
+package org.usfirst.frc.team467.robot;
+
+import org.apache.log4j.Logger;
+
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.MotorSafetyHelper;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+public class Elevator {
+	private static Elevator instance;
+	private static final Logger LOGGER = Logger.getLogger(Elevator.class);
+
+	private AnalogInput heightSensor;
+	private WPI_TalonSRX heightController;
+	private double feetPerTick;
+	private int maxTicksPerIteration;
+	private double previousHeight;
+	private MotorSafetyHelper m_safetyHelper = new MotorSafetyHelper(heightController);
+	
+	private Stops targetHeight;
+	
+	public enum Stops {
+		noStop(RobotMap.ELEVATOR_MIN_HEIGHT_IN_FEET),
+		min(RobotMap.ELEVATOR_MIN_HEIGHT_IN_FEET),
+		fieldSwitch(2),
+		lowScale(6),
+		highScale(8);
+		
+		/**
+		 * Height in feet
+		 */
+		public final double height;
+		
+		Stops(double height) {
+			this.height = height;
+		}
+	}
+
+	private Elevator() {
+		heightSensor = new AnalogInput(RobotMap.ELEVATOR_HEIGHT_SENSOR_ID);
+		heightController = new WPI_TalonSRX(RobotMap.ELEVATOR_MOTOR_CHANNEL);
+		targetHeight = Stops.noStop;
+		feetPerTick = (RobotMap.GEAR_CIRCUMFERENCE_IN_INCHES / 12) / RobotMap.ELEVATOR_TICKS_PER_TURN;
+		maxTicksPerIteration = RobotMap.ELEVATOR_TICKS_PER_TURN * RobotMap.MAX_ELEVATOR_RPM / 60 / 100; // 10 ms per iteration
+		previousHeight = getHeight();
+	}
+	
+	/**
+	 * The lowest value is 196.0, the maximum value is 3741.0. The middle is 1968.5
+	 * New max: 2980, new min:956.5
+	 * 16.9 ticks = 1 inch
+	 * 1 rotation=253 ticks
+	 * 
+	 * @return a single instance of the Elevator object.
+	 */
+	public static Elevator getInstance() {
+		if (instance == null) {
+			instance = new Elevator();
+		}
+		return instance;
+	}
+
+	/**
+	 * Moves based on the Xbox controller analog input
+	 * 
+	 * @param speed The velocity. Shall be a value between -1 and 1.
+	 */
+	public void manualMove(double speed) {
+
+		targetHeight = Stops.noStop;
+		if (m_safetyHelper != null) {
+			m_safetyHelper.feed();
+		}
+		if (isOutOfRange()) {
+			heightController.set(0);
+			//DriverStation.getInstance().setDriverRumble(0.5);
+			return; // Don't bother with any more logic here.
+		}
+		
+		double currentHeight = getHeight();
+		for (Stops stop : Stops.values()) {
+			if ((previousHeight < stop.height && currentHeight >= stop.height)
+				|| (previousHeight > stop.height && currentHeight <= stop.height)) {
+				
+				//DriverStation.getInstance().setDriverRumble(0.5);
+			} else {
+				//DriverStation.getInstance().setDriverRumble(0.0);
+			}
+		}
+		if(currentHeight >= 2 && currentHeight <= 2.1 || currentHeight >= 4 && currentHeight <=4.1 || currentHeight >= 6 && currentHeight <= 6.1)
+			LOGGER.debug("Rumbling right now");
+			else {		
+			}
```

Comment:
Is this "if" block of code obsolete now, because it does almost the same thing as the for loop with stops above?